### DISCUSSION TITLE
TechPrep - Tech Interview preparation app

### DISCUSSION THREAD

**Discussion by anburocky3** [anburocky3/anburocky3#3] [Ideas]
**Problem:**
There are lots of websites that has interview questions in unorganized manner. The problem I faced is that, the questions are clumsy and evaluating myself is somewhat difficult.

**Solution:**
Bringing a solution where we can able to have all interview questions based on levels (Easy, Medium, Hard) on organized and UX manner. We should have more activities to make the guest feel confident.

**Activities includes:**
1. `QAs block` which should have image embedding support.
2. Quiz kind of activity where they can attend and evaluate themself.

**Curriculam workflow:**
Show have the ability to pick the interview based on the following.
1. JobWise (e.g. Frontend Developer)
2. TechWise (e.g. Javascript)
3. TopicWise (e.g. JS Arrays, DSA, etc.)
4. OrganizationWise (e.g. Zoho)
5. Commonly mentioned Problems (e.g. write a program...)

**Quiz strategry:**

1. Show scores after the quiz.
2. Show the wrong selected with the correct answer.

**Quiz supported types:**
1. Multi option quiz
2. Fill in the blanks

**Additional features:**
1. Should have a functionality to know the roadmap of each modules.
6. Interview rounds information

If you have any other, mention here:

References:
1. https://prepinsta.com
3. https://www.interviewbit.com
7. https://www.theodinproject.com

**Comment by yrd369** [comment]
Many companies follow unique pattern questions. we can schedule questions based on company, like if a person want to prepare for a specific company interview, they can search the questions based on company's name.

**Reply by anburocky3** [reply]
How we will get those questions? Some will not reveal it right?

**Reply by yrd369** [reply]
we can get question patterns, for example if we take accenture, there will be 3 rounds, we have to show how many rounds of interview, whether its aptitude based or technical oriented if possible we can also share previously asked questions.

**Comment by sharif-22** [comment]
- We can add questions for specific topics such as arrays, DOM manipulation, OOP, DSA, Frontend, Backend, etc.
- Additionally, we can implement quizzes based on topics and endless stream of questions for continuous learning.

**Comment by Jeya-rosini** [comment]
The problem I faced during preparation with interview questions are many website only give theory based questions

Example: 
1) what is difference between var and let? 
2) what is OOPS? 

But the main problem is in logical part, so I suggest to have a logical based questions. 

> **Mainly question based on writing a code to get particular alignment of words, shape and structure**

> **write a code for simple mathematics and equation**

> **For example : write a code simple filter, sort, reduce and more array methods - in JS Interview questions**

We can have a separate section for practices the code by writing on their own
